k8scrdclient/k8scrdclient.go

```diff
@@ -153,3 +164,38 @@ func (c *CRDClient) ensureStatusSubresourceCreated(ctx context.Context, crd *api
 
 	return nil
 }
+
+func crdAPIVersionEqual(a, b *apiextensionsv1beta1.CustomResourceDefinition) bool {
+	if a != nil && b != nil && a.TypeMeta.APIVersion == b.TypeMeta.APIVersion {
+		return true
+	}
+
+	return false
+}
+
+func crdStatusEqual(a, b *apiextensionsv1beta1.CustomResourceDefinition) bool {
+	if a != nil && b != nil && a.Spec.Subresources == nil && b.Spec.Subresources == nil {
+		return true
+	}
+
+	if a != nil && b != nil &&
+		a.Spec.Subresources != nil && b.Spec.Subresources != nil &&
+		a.Spec.Subresources.Status != nil && b.Spec.Subresources.Status != nil &&
+		a.Spec.Subresources.Status.String() == b.Spec.Subresources.Status.String() {
+		return true
+	}
```

Comment:
Form the conditions above it looks to me that if `a == nil || b == nil` then it's always false. I would move that up. Or even move that up to `ensureUpdated` before all those method calls. Also til looks that when `a == nil && b == nil` it returns false which seem incorrect to me.

Reply:
Rewrote it a bit. It is a bit 50 50 to me so not sure if it is really better. Doing this in a pretty way is somehow not possible.

Reply:
You can continue with the nil checks like on the top:

```
	if a.Spec.Subresources == nil && b.Spec.Subresources == nil {
		return true
	}

	if a.Spec.Subresources != nil || b.Spec.Subresources != nil {
		return false
	}

	if a.Spec.Subresources.Status == nil && b.Spec.Subresources.Status == nil {
		return true
	}

	if a.Spec.Subresources.Status != nil || b.Spec.Subresources.Status != nil {
		return false
	}

	if a.Spec.Subresources.Status.String() == b.Spec.Subresources.Status.String() {
		return true
	}
```

Reply:
I am not sure if all of this makes anything better at this point. I consider it a matter of taste. What is right now is good enough considering our requirements. If you think that should all be different I would gladly let you take the driver seat here. ❤️

Reply:
Yeah not a big deal.